File: HardwareCode/states.js

var IDLE = {
  id: 0,
  master: {
    values: {
      
    },
    events: {
      
    }
  },
  tablet: {
    values: {
      
    },
    events: {
      
    }
  }
};
var STATE_IDLE = 0;
var PISTON = {
  id: 1,
  master: {
    values: {
      encodedChord: new HardwareValue(1, 0, Manager.TYPE_UINT32)
    },
    events: {
      playFChord: function playFChord() { manager.sendEvent(0, 1); },
      playBbChord: function playBbChord() { manager.sendEvent(1, 1); },
      playD5Chord: function playD5Chord() { manager.sendEvent(2, 1); },
      playAbChord: function playAbChord() { manager.sendEvent(3, 1); },
      playDbChord: function playDbChord() { manager.sendEvent(4, 1); },
      playCChord: function playCChord() { manager.sendEvent(5, 1); },
      playeChord: function playeChord() { manager.sendEvent(6, 1); },
      playaChord: function playaChord() { manager.sendEvent(7, 1); },
      playDChord: function playDChord() { manager.sendEvent(8, 1); },
      playGChord: function playGChord() { manager.sendEvent(9, 1); },
      playAChord: function playAChord() { manager.sendEvent(10, 1); },
      playC_InvertedChord: function playC_InvertedChord() { manager.sendEvent(11, 1); },
      playF_InvertedChord: function playF_InvertedChord() { manager.sendEvent(12, 1); },
      retractAll: function retractAll() { manager.sendEvent(13, 1); },
      engageMuteBar: function engageMuteBar() { manager.sendEvent(14, 1); },
      disengageMuteBar: function disengageMuteBar() { manager.sendEvent(15, 1); }
    }
  },
  tablet: {
    values: {
      
    },
    events: {
      
    }
  }
};
var STATE_PISTON = 1;

var STATES = {
  IDLE: IDLE,
  PISTON: PISTON
};
var manager = new Manager([IDLE, PISTON]);

File: HardwareCode/GUITARAMIB1/states.cpp
#include "states.h"

static const StateInfo state_infos[2] = {
  {IDLE::setup, IDLE::enter, IDLE::exit, IDLE::loop, IDLE::event},
  {PISTON::setup, PISTON::enter, PISTON::exit, PISTON::loop, PISTON::event}
};

static const WireValue wire_values[1] = {
  {1, 0, sizeof(uint32_t), (Value<void*>*) &PISTON::encodedChord}
};

MasterManager<State, 2, 1> manager(0xc41e52b9, state_infos, wire_values, 0);

namespace IDLE {


void event(uint8_t ev) {
  switch (ev) {
  
  default:
    break;
  }
}


}
namespace PISTON {
Value<uint32_t> encodedChord;

void event(uint8_t ev) {
  switch (ev) {
  case 0:
    events::playFChord();
    break;
  case 1:
    events::playBbChord();
    break;
  case 2:
    events::playD5Chord();
    break;
  case 3:
    events::playAbChord();
    break;
  case 4:
    events::playDbChord();
    break;
  case 5:
    events::playCChord();
    break;
  case 6:
    events::playeChord();
    break;
  case 7:
    events::playaChord();
    break;
  case 8:
    events::playDChord();
    break;
  case 9:
    events::playGChord();
    break;
  case 10:
    events::playAChord();
    break;
  case 11:
    events::playC_InvertedChord();
    break;
  case 12:
    events::playF_InvertedChord();
    break;
  case 13:
    events::retractAll();
    break;
  case 14:
    events::engageMuteBar();
    break;
  case 15:
    events::disengageMuteBar();
    break;
  default:
    break;
  }
}


}

